content/developer/reference/backend/security.rst

```diff
@@ -481,3 +515,78 @@ field value can be easily achieved safely:
 
 The above method is obviously still too optimistic and additional verifications
 on the record id and field value must be done.
+
+
+Injectable Request
+---------------------------
+
+Some may need to make an HTTP request directly, it is require to follow some best practices
+
+**Injectable domain**
+
+Care must be taken while manipulating the domain name of the request, the domain should never depend on user input.
+If the domain can be changed by the user, a malicious user could use the server running the code to send multiple requests to a chosen target.
+This is known as a reflective-DDOS attack, where one can overload a target without sending the request himself.
+
+**URL path/query Injectable**
+
+In the same way as for the domain, care must be taken when manipulating the string of the URL. It should not allow for any user input to be added to the URL. If a user were to modify the URL, he could make the request do other actions than the expected one.
+
+**Timeout**
+
+By default, requests never timeout. If the destination server is not reachable, the running working will halt forever.
+For this reason, it is crucial to set a timeout for the request.
+It is best practice to set the timeout slightly higher than a multiple of 3, which is the default retransmission window of TCP.
+
+**Error Handling**
+
+Errors should always be handled explicitly, an error response code can happen depending on the state of the other host. In order to avoid a simple crash with traceback presented to the user, you should verify
+response.ok is set to True.
+
+
+Automated Testing
+=================
+
+
+SQL Injection detection
+---------------------------
+
+Detection of potential SQL Injection vulnaribility is handled by a custom module of pylint.
+This test decompose the AST of the python in order to infer the value of the string that is used as a SQL query.
+
+You should never use the query string to pass a parameter. As seen in the section of Security pitfall,
+
+
+.. code-block:: python
+
+    # this is not allowed
+    def get_table(self, arg1):
+        self.env.cr.execute("SELECT * FROM table WHERE %s" % arg1)
+
+    # this is allowed
+    def get_table(self, arg1):
+        self.env.cr.execute("""SELECT * FROM table WHERE %s""", (arg1,))
```

Comment:
Our own coding guideline at the ORM is to pass the arguments in a list, we find it more readable. It is a slight adaptation of python's DB2 API (which takes a tuple, not a list) but much more readable for 1 argument 

```suggestion
        self.env.cr.execute("""SELECT * FROM table WHERE %s""", [arg1])
```